pkg/syncer/fullsync.go

```diff
@@ -1608,102 +1735,26 @@ func cleanupUnusedPVCsAndSnapshotsFromGuestCluster(ctx context.Context) {
-					}
-				}
-			}
+		if vs.ObjectMeta.DeletionTimestamp != nil {
+			RemoveCNSFinalizerFromSnapIfTKGClusterDeleted(ctx, snapshotterClient, &vs,
+				cnsoperatortypes.CNSSnapshotFinalizer, false)
 		}
 	}
 
-	// Get all PVCs and check if any marked for deletion but have finalizer "cns.vmware.com/pvc-protection",
+	// Get all PVCs and check if any marked for deletion but have finalizer "cns.vmware.com/pvc-delete-protection",
 	// set while creation from guest cluster.
-	// If found, remove the finalizer "cns.vmware.com/pvc-protection" from that PVC.
+	// If found, remove the finalizer "cns.vmware.com/pvc-delete-protection" from that PVC.
 	pvcList, err := k8sClient.CoreV1().PersistentVolumeClaims("").List(ctx, metav1.ListOptions{})
 	if err != nil {
 		log.Errorf("cleanupUnusedPVCsAndSnapshotsFromGuestCluster: failed to list PersistentVolumeClaims."+
 			" Err: %v", err)
 		return
 	}
 	for _, pvc := range pvcList.Items {
-		// Check if PVC is being deleted and has "cns.vmware.com/pvc-protection" finalizer
-		if (pvc.ObjectMeta.DeletionTimestamp != nil) &&
-			(len(pvc.ObjectMeta.Finalizers) != 0) {
-			for i, finalizer := range pvc.ObjectMeta.Finalizers {
-				if finalizer != cnsoperatortypes.CNSVolumeFinalizer {
-					continue
-				}
-				// Fetch guest cluster name from PVC label and check if that guest cluster,
-				// where associated PVC was created, is running or not.
-				var tkcClusterName string
-				for key := range pvc.ObjectMeta.Labels {
-					if strings.Contains(key, "TKGService") {
-						tkcDetails := strings.Split(key, "/")
-						tkcClusterName = tkcDetails[0]
-						break
-					}
-				}
-				cc := &ccV1beta1.Cluster{}
-				err := ccClient.Get(ctx, client.ObjectKey{
-					Namespace: pvc.Namespace,
-					Name:      tkcClusterName,
-				}, cc)
-				if err != nil && !apierrors.IsNotFound(err) {
-					msg := fmt.Sprintf("cleanupUnusedPVCsAndSnapshotsFromGuestCluster: failed to get Cluster %q "+
-						"in %q namespace. Err: %+v", tkcClusterName, pvc.Namespace, err)
-					log.Error(msg)
-				} else if ((err == nil) && (cc.Status.Phase != "Running")) || apierrors.IsNotFound(err) {
-					// Remove finalizer if associated guest cluster is not-running/deleted already
-					log.Infof("cleanupUnusedPVCsAndSnapshotsFromGuestCluster: Removing %q finalizer from PVC "+
-						"with name: %q on namespace: %q in Terminating state",
-						cnsoperatortypes.CNSVolumeFinalizer, pvc.Name, pvc.Namespace)
-					pvc.ObjectMeta.Finalizers = slices.Delete(pvc.ObjectMeta.Finalizers, i,
-						i+1)
-					_, err = k8sClient.CoreV1().PersistentVolumeClaims(pvc.Namespace).Update(ctx,
-						&pvc, metav1.UpdateOptions{})
-					if err != nil {
-						msg := fmt.Sprintf("cleanupUnusedPVCsAndSnapshotsFromGuestCluster: failed to update "+
-							"supervisor PVC %q in %q namespace. Err: %+v", pvc.Name, pvc.Namespace, err)
-						log.Error(msg)
-					}
-				}
-			}
+		// Check if PVC is being deleted and has "cns.vmware.com/pvc-delete-protection" finalizer
```

Comment:
Can you update this comment and mention, finalizer will only be removed in 2 cases 

1) namespace is deleted
2) tkc cluster from which this PVC is created is deleted.